.github/actions/triggerWorkflowAndWait/triggerWorkflowAndWait.js

```diff
@@ -0,0 +1,138 @@
+const _ = require('underscore');
+const core = require('@actions/core');
+const github = require('@actions/github');
+const ActionUtils = require('../../libs/ActionUtils');
+const GithubUtils = require('../../libs/GithubUtils');
+const promiseWhile = require('../../libs/promiseWhile');
+
+/**
+ * The maximum amount of time we'll wait for a new workflow to start after sending the workflow_dispatch event.
+ * @type {number}
+ */
+const NEW_WORKFLOW_TIMEOUT = 120000;
+
+/**
+ * The rate at which we'll poll the GitHub API to check for workflow status changes.
+ * @type {number}
+ */
+const POLL_RATE = 10000;
+
+const run = function () {
+    const octokit = github.getOctokit(core.getInput('GITHUB_TOKEN', {required: true}));
+    const githubUtils = new GithubUtils(octokit);
+    const workflow = core.getInput('WORKFLOW', {required: true});
+    const inputs = ActionUtils.getJSONInput('INPUTS', {required: false}, {});
+
+    console.log('This action has received the following inputs: ', {workflow, inputs});
+
+    if (_.keys(inputs).length > 10) {
+        const err = new Error('Inputs to the workflow_dispatch event cannot have more than 10 keys, or GitHub will 🤮');
+        console.error(err.message);
+        core.setFailed(err);
+        process.exit(1);
+    }
+
+    // GitHub's createWorkflowDispatch returns a 204 No Content, so we need to:
+    // 1) Get the last workflow run
+    // 2) Trigger a new workflow run
+    // 3) Poll the API until a new one appears
+    // 4) Then we can poll and wait for that new workflow run to conclude
+    let previousWorkflowRunID;
+    let newWorkflowRunID;
+    let hasNewWorkflowStarted = false;
+    let workflowCompleted = false;
+    return githubUtils.getLatestWorkflowRunID(workflow)
+        .then((lastWorkflowRunID) => {
+            console.log(`Latest ${workflow} workflow run has ID: ${lastWorkflowRunID}`);
+            previousWorkflowRunID = lastWorkflowRunID;
+
+            console.log(`Dispatching workflow: ${workflow}`);
+            return octokit.actions.createWorkflowDispatch({
+                owner: GithubUtils.GITHUB_OWNER,
+                repo: GithubUtils.EXPENSIFY_CASH_REPO,
+                workflow_id: workflow,
+                ref: 'main',
+                inputs,
+            });
+        })
+
+        .catch((err) => {
```

Comment:
Any reason this catch isn't at the end of the chain?

Reply:
Yeah, it's mostly for logging purposes. If we fail this early on in the chain, then we know that we failed to dispatch a new workflow in the first place. If we fail after this catch block, we know that we did successfully dispatch a new workflow, but that it ended with a non-successful conclusion.